### PR TITLE
Milestone 2: Math magicans(tests)

### PR DESCRIPTION
**Pull request by Gardimy**
**For this milestone, I write unit tests for the Math Magicians app, using the Jest and React Testing Library Libraries.**

### CODE REVIEW

**Review by AmaduKamara — COMMENTED** [review]
# Status: Invalid Submission ❌ ❌ ❌

Hi @Gardimy,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that does not follow the proper requirements as expected:

- [ ] It is required that every member contribute to this project and commit history will be checked to see if everyone contributes. Only one person made commits to this project.
- [ ] Kindly have your team member participate in this project or if there is a proof of absence or allowed to work solo by student success, kindly attach reference in your PR. 👍🏽

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._
```

File: src/test/operate.test.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import '@testing-library/jest-dom/extend-expect';
+import operate from '../logic/operate';
+
```
Comment (ShadyShawkat) [inline src/test/operate.test.js:3]: - [ ] Please make sure to implement more integration tests for user interactions, as per [this](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/project_tests.md#:~:text=Use%20React%20Testing%20Library%20to%20simulate%20user%20interaction.) requirement.

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @Gardimy, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- All linters are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by jerryowusu — APPROVED** [review]
Hi Team

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

chelsea-mason-approve


## Highlights
All tests passed :+1: 
Great effort in making the requested changes :+1: 

## Optional suggestions

- [ ] Nice work so far. However, you were suppose to write unit tests for all the components  as part of the project requirements. I am making it optional because the previous reviewer did not stress on that, but I would strongly advise you to work on that before moving to the next milestone. :pray: 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**